### PR TITLE
[CFT Custom fields]: Show error message after pressing Add or Add another

### LINKED ISSUE #46610: [CFT Custom fields]: Show error message after pressing Add or Add another
When adding a new custom field and tabbing to the value field, the “The value is required” validation error immediately appears. The error should only appear after clicking `Add` or `Add another`.

screenshot-2024-04-14-at-07 48 07

### LINKED ISSUE #46611: [CFT Custom fields]: Allow value to be edited on unsaved new products
On an unsaved new product, values cannot be edited.

Screen Recording on 2024-04-15 at 05-45-46

### LINKED ISSUE #46613: [CFT Custom fields]: Trim whitespaces of custom field names and values
In the classic editor, whitespace at the beginning and end of custom field names and values are trimmed.
In the new editor, the whitespace is kept, which is error-prone.

Screen Recording on 2024-04-15 at 06-01-07

### PR DESCRIPTION
**Pull request by mdperez86**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #46610
Closes #46611
Closes #46613

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure feature `New product editor` is enabled under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`
2. Make sure feature `product-custom-fields` is enabled under `Features` tab from `/wp-admin/tools.php?page=woocommerce-admin-test-helper` (WooCommerce Beta Tester plugin) is required
3. Go to `Products` -> `Add new` from the left menu to create a new product
4. Then go to the `Organization` tab
5. At the end of the page a new `Show custom fields` toggle should be shown
6. The following should be fixed.
   - [x] When adding a new custom field and tabbing to the value field, the “The value is required” validation error immediately appears. The expected behavior is: when we tab to the field for the first time, we shouldn't show any error image
   - [x] On an unsaved new product, values cannot be edited. The expected behavior is: to be able to edit the value in anytime Screen Recording on 2024-04-15 at 05-45-46
   - [x] In the classic editor, whitespace at the beginning and end of custom field names and values are trimmed.
In the new editor, the whitespace is kept, which is error-prone. The expected behavior is: to trim the values introduced in the field image
 

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: packages/js/product-editor/src/components/custom-fields/custom-fields.tsx [hunk 1]
```diff
@@ -163,14 +163,16 @@ export function CustomFields( {
...
 			{ showCreateModal && (
 				<CreateModal
+					values={ customFields }
 					onCreate={ handleCreateModalCreate }
 					onCancel={ handleCreateModalCancel }
 				/>
 			) }
 
-			{ selectedCustomField && (
+			{ selectedCustomFieldIndex != undefined && (
```
Comment (octaedro) [inline packages/js/product-editor/src/components/custom-fields/custom-fields.tsx]: I think that this should be `!==` instead of `!=`

**Review by octaedro — COMMENTED** [review]
Good job @mdperez86! This is testing well here and the code looks good. I only left a small comment. After addressing that I think this PR will be ready to merge.

File: packages/js/product-editor/src/components/custom-fields/custom-fields.tsx (hunk 1 above)
Comment (mdperez86) [reply packages/js/product-editor/src/components/custom-fields/custom-fields.tsx]: Thanks @octaedro done -> [01f489e](https://github.com/woocommerce/woocommerce/pull/46703/commits/01f489e92e50177258a2355b645ee24d2b0e61ae)

**Review by octaedro — APPROVED** [review]
LGTM! 🚀

**Pull request merged**